Automattic/sensei
# Add Lesson Video Block

Fixes #5608 

### Changes proposed in this Pull Request

* Adds a Lesson Video block for rendering lesson video in the templates
* Disables legacy lesson video hook rendering the `_lesson_video` video
* Removes featured video block from content when using a template that has `` block.

### Testing instructions

- In settings, under appearance, select a video template.
- Go to a lesson in the editor and add a featured video block.
- In the `lesson.html` template of the one selected in settings, replace `<p class="has-text-align-center" style="font-size:100px">▶</p>` with `` (this is the part that displays the featured video - in sensei-pro).
- Go to the lesson on the frontend and see the featured video displayed.


### TODO

- [x] Figure out best way to render the video

- [x] Try for consistent styling for supported providers (maybe in a different issue)

- [x] Compatibility with interactive video blocks

- [x] Disable legacy lesson video hook rendering the _lesson_video video

- [x] Find a way to hide the the featured video block in the post content.
